spec/lib/oauth/client_credentials_request_spec.rb

```diff
@@ -59,6 +60,26 @@ module Doorkeeper::OAuth
         expect(token_creator).to receive(:create).with(client, Doorkeeper::OAuth::Scopes.from_string('email'))
         subject.authorize
       end
+
+      context 'and limited-permissions client' do
+        let(:client_scopes) { Doorkeeper::OAuth::Scopes.from_string('public phone') }
+
+        before do
+          allow(application).to receive(:scopes).and_return(client_scopes)
+        end
+
+        it 'issues an access token with the intersection of default scopes and client scopes if none was requested' do
+          expect(token_creator).to receive(:create).with(client, Doorkeeper::OAuth::Scopes.from_string('public'))
+          subject.authorize
+        end
+
+        it 'issues an access token with requested scopes if the client has those permissions' do
+          subject = ClientCredentialsRequest.new(server, client, scope: 'phone')
+          subject.issuer = token_creator
+          expect(token_creator).to receive(:create).with(client, Doorkeeper::OAuth::Scopes.from_string('phone'))
+          subject.authorize
+        end
```

Comment:
I just copied from above existing specs. it expects `:create` will be called with specific arguments (during the following statements), so `subject.authorize` must go after this expectation.